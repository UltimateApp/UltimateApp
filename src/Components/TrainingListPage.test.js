import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { render, fireEvent, cleanup } from 'react-native-testing-library';
import store from '../Store/testStore';
import fixtures from '../Fixtures/TestFixtures';

import ConnectedTrainingListPage, { TrainingListPage } from './TrainingListPage';

afterEach(cleanup);

describe('<TrainingListPage />', () => {
  it('renders correctly when connected', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ConnectedTrainingListPage />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('renders correctly with a training', () => {
    const tree = renderer
      .create(<TrainingListPage allDrills={fixtures.drills} trainings={fixtures.trainings} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('links to training page', async () => {
    const training = fixtures.trainings[0];
    const navigation = { navigate: jest.fn() };
    const { getByText } = render(
      <TrainingListPage navigation={navigation} allDrills={fixtures.drills} trainings={fixtures.trainings} />,
    );

    await fireEvent.press(getByText(fixtures.trainings[0].title));

    expect(navigation.navigate).toBeCalledWith('TrainingPage', { training });
  });
});
