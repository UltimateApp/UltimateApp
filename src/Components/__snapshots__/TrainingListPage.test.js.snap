// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrainingListPage /> renders correctly when connected 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingLeft": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A6A6A6",
        "marginBottom": 20,
      }
    }
  >
    1
     training sessions available
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget",
          "drills": Array [
            2,
            3,
          ],
          "id": 1,
          "image": "https://d3j2bju5c7tc02.cloudfront.net/2016_44/backhand.jpg",
          "source": "Rise Up",
          "title": "Dump & Swing",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": 80,
              "marginBottom": 20,
              "opacity": 1,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(0,0,0)",
                "borderRadius": 5,
                "height": 80,
                "justifyContent": "center",
                "marginRight": 10,
                "padding": 10,
                "width": 80,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://d3j2bju5c7tc02.cloudfront.net/2016_44/backhand.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 80,
                    "width": 80,
                  },
                  Object {
                    "borderRadius": 5,
                    "opacity": 0.5,
                  },
                ]
              }
            />
            <Text
              style={
                Object {
                  "color": "#e5e5e5",
                  "fontSize": 14,
                  "textAlign": "center",
                }
              }
            >
              40
               min
            </Text>
            <Text
              style={
                Object {
                  "color": "#e5e5e5",
                  "fontSize": 14,
                  "textAlign": "center",
                }
              }
            >
              2
              + players
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 5,
                "paddingBottom": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Rise Up
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Dump & Swing
            </Text>
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Legs, Defense
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;

exports[`<TrainingListPage /> renders correctly with a training 1`] = `
<View
  style={
    Object {
      "backgroundColor": "#fff",
      "height": "100%",
      "paddingLeft": 20,
      "paddingTop": 10,
    }
  }
>
  <Text
    style={
      Object {
        "color": "#A6A6A6",
        "marginBottom": 20,
      }
    }
  >
    1
     training sessions available
  </Text>
  <RCTScrollView
    data={
      Array [
        Object {
          "description": "Lorem ipsum dolor sit amet, consectetur adipiscing elit. Donec scelerisque dignissim felis, at tristique mi efficitur a. Nulla pellentesque odio nunc, ut sodales ex pulvinar at. In hac habitasse platea dictumst. Pellentesque sodales nisl lorem, ac lacinia nisl lacinia eu. Proin lacinia viverra mauris, et pharetra ipsum tempus eget. Ut maximus sapien in hendrerit ultricies. Maecenas vel diam tincidunt, consequat nunc a, mattis eros. Curabitur a eros a nulla malesuada gravida. Donec massa quam, suscipit eget efficitur et, euismod sed nulla. Morbi mattis et magna a aliquam. Fusce pellentesque vel erat eget",
          "drills": Array [
            2,
            3,
          ],
          "id": 1,
          "image": "https://d3j2bju5c7tc02.cloudfront.net/2016_44/backhand.jpg",
          "source": "Rise Up",
          "title": "Dump & Swing",
        },
      ]
    }
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    numColumns={1}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View>
      <View
        onLayout={[Function]}
        style={null}
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "flexDirection": "row",
              "height": 80,
              "marginBottom": 20,
              "opacity": 1,
            }
          }
        >
          <View
            accessibilityIgnoresInvertColors={true}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgb(0,0,0)",
                "borderRadius": 5,
                "height": 80,
                "justifyContent": "center",
                "marginRight": 10,
                "padding": 10,
                "width": 80,
              }
            }
          >
            <Image
              source={
                Object {
                  "uri": "https://d3j2bju5c7tc02.cloudfront.net/2016_44/backhand.jpg",
                }
              }
              style={
                Array [
                  Object {
                    "bottom": 0,
                    "left": 0,
                    "position": "absolute",
                    "right": 0,
                    "top": 0,
                  },
                  Object {
                    "height": 80,
                    "width": 80,
                  },
                  Object {
                    "borderRadius": 5,
                    "opacity": 0.5,
                  },
                ]
              }
            />
            <Text
              style={
                Object {
                  "color": "#e5e5e5",
                  "fontSize": 14,
                  "textAlign": "center",
                }
              }
            >
              40
               min
            </Text>
            <Text
              style={
                Object {
                  "color": "#e5e5e5",
                  "fontSize": 14,
                  "textAlign": "center",
                }
              }
            >
              2
              + players
            </Text>
          </View>
          <View
            style={
              Object {
                "padding": 5,
                "paddingBottom": 10,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Rise Up
            </Text>
            <Text
              style={
                Object {
                  "flex": 3,
                  "flexWrap": "wrap",
                  "fontSize": 20,
                  "fontWeight": "bold",
                }
              }
            >
              Dump & Swing
            </Text>
            <Text
              style={
                Object {
                  "color": "#A6A6A6",
                  "flex": 2,
                  "fontSize": 14,
                }
              }
            >
              Legs, Defense
            </Text>
          </View>
        </View>
      </View>
    </View>
  </RCTScrollView>
</View>
`;
